### PR TITLE
Verify that the user jwts has hca group claim

### PR DESCRIPTION
**Pull request by aaclan-ebi**

### CODE REVIEW

File: src/main/java/org/humancellatlas/ingest/security/User.java [hunk 1]
```diff
@@ -0,0 +1,18 @@
+package org.humancellatlas.ingest.security;
+
+import com.auth0.jwt.JWT;
+import com.auth0.jwt.interfaces.DecodedJWT;
+import com.auth0.spring.security.api.authentication.JwtAuthentication;
+
+public class User {
```
Comment (rolando-ebi) [inline src/main/java/org/humancellatlas/ingest/security/User.java]: I think `UserJwt` would be a clearer name for this class

File: .travis.yml [hunk 2]
```diff
@@ -17,7 +17,8 @@ env:
   - USR_AUTH_AUDIENCE="https://dev.data.humancellatlas.org/"
   - SVC_AUTH_AUDIENCE="https://dev.data.humancellatlas.org/"
   - AUTH_ISSUER="https://humancellatlas.auth0.com/"
-  - SECURITY_WHITELIST_DOMAINS="ebi.ac.uk"
```
Comment (MightyAx) [inline .travis.yml]: Just to confirm: Do we no longer need the `SECURITY_WHITELIST_DOMAINS` environment variable?

Comment (aaclan-ebi) [reply .travis.yml]: yup, checking the group claim in the token is set to `hca` should be sufficient. It wouldn't get set as `hca` if it doesn't belong to the whitelist. The whitelist is being maintained in the HCA Auth0 account which Fusillade/DCP Auth Service is using for authentication

Comment (justincc) [reply .travis.yml]: I presume this would have been wrong anyway since domains other than `ebi.ac.uk` should be whitelisted? e.g. broad, czi, etc.

**Comment by aaclan-ebi** [comment]
Reverted running of tests in docker build as it has a dependency on Mongo service.

File: .travis.yml (hunk 2 above)
Comment (aaclan-ebi) [reply .travis.yml]: yeah, and there might be more. The advantage of checking the group claim in the token is that we won't have to maintain that whitelist as well.

**Pull request merged**